### PR TITLE
feature: sample component test

### LINKED ISSUE #343: Creation sample quality assurance
**Describe the problem**
Examples in bff do not have a component testing area

**Expected behavior**
Creation of an area for testing beagle components in bff.

### PR DESCRIPTION
**Pull request by luisoliveirazup**
closes #343  
Creation of an area for testing beagle components in bff.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: iOS/Example/BeagleDemo/BeagleDemo/BeagleStyle.swift [hunk 1]
```diff
@@ -30,8 +30,11 @@ struct AppTheme {
         .FORM_SUBMIT_STYLE: Self.formButton,
         .NAVIGATION_BAR_GREEN_STYLE: Self.designSystemStyleNavigationBar,
         .NAVIGATION_BAR_DEFAULT_STYLE: Self.designSystemStyleNavigationBarDefault,
+        .TEXT_INPUT_STYLE: Self.designSystemTextInput,
         .TAB_VIEW_STYLE: Self.tabView,
-        .TEXT_INPUT_STYLE: Self.designSystemTextInput
+        .DESIGN_SYSTEM_STYLE_BUTTON_SCREEN_BUTTON: designSystemScreenButton,
+        .STYLE_TAB_VIEW_BFF: tabViewBff,
```
Comment (botbeaglezup) [inline iOS/Example/BeagleDemo/BeagleDemo/BeagleStyle.swift:36]: <table data-meta="generated_by_ios">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/f4a2a119c1ae51d106f7005e730420709ec9f849/iOS/Example/BeagleDemo/BeagleDemo/BeagleStyle.swift#L36"></span>Collection literals should not have trailing commas.<br />
<code>trailing_comma</code> <code>BeagleStyle.swift:36</code></td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/f4a2a119c1ae51d106f7005e730420709ec9f849/iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift#L62">iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift#L62</a> - Prefer object literals over image and color inits.<br />
<code>object_literal</code> <code>Constants.swift:62</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/f4a2a119c1ae51d106f7005e730420709ec9f849/iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift#L65">iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift#L65</a> - Prefer object literals over image and color inits.<br />
<code>object_literal</code> <code>Constants.swift:65</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/f4a2a119c1ae51d106f7005e730420709ec9f849/iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift#L68">iOS/Example/BeagleDemo/BeagleDemo/Constants/Constants.swift#L68</a> - Prefer object literals over image and color inits.<br />
<code>object_literal</code> <code>Constants.swift:68</code></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_ios">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/sample/src/main/res/values/colors.xml [hunk 2]
```diff
@@ -24,4 +24,7 @@
     <color name="colorGray">#6F6F6F</color>
     <color name="colorKaie">#1b2534</color>
```
Comment (botbeaglezup) [inline android/sample/src/main/res/values/colors.xml:25]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle/blob/f4a2a119c1ae51d106f7005e730420709ec9f849/android/sample/src/main/res/values/colors.xml#L25"></span>The resource <code>R.color.colorKaie</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle/blob/f4a2a119c1ae51d106f7005e730420709ec9f849/android/sample/src/main/res/values/styles.xml#L128">android/sample/src/main/res/values/styles.xml#L128</a> - The resource <code>R.style.DesignSystem_Text_NetworkImage</code> appears to be unused</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ClickActionScreenBuilder.kt [hunk 3]
```diff
@@ -26,7 +26,7 @@ import br.com.zup.beagle.widget.layout.Screen
 import br.com.zup.beagle.widget.layout.ScreenBuilder
 import br.com.zup.beagle.widget.ui.Text
 
-object ClickActionScreenBuilder : ScreenBuilder {
+object ClickActionScreenBuilder: ScreenBuilder {
```
Comment (LG95) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ClickActionScreenBuilder.kt:29]: No `qaFlag` for this sample?

Comment (luisoliveirazup) [reply backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ClickActionScreenBuilder.kt:29]: This object returns a result screen of clicking action on beagle components.

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ComponentScreenBuilder.kt [hunk 4]
```diff
@@ -82,14 +100,16 @@ object ComponentScreenBuilder : ScreenBuilder {
...
-                createMenu("Custom Platform", CUSTOM_PLATFORM_SAMPLE_ENDPOINT)
+                createMenu("Custom Platform", CUSTOM_PLATFORM_SAMPLE_ENDPOINT),
+                createMenu("Safe Area", SCREEN_SAFE_AREA),
+                createMenu("Flex",SCREEN_FLEX_ENDPOINT)
             )
         )
     )
 
-    private fun createMenu(text: String, path: String) = Button(
+    private fun createMenu(text: String, path: String, qaFlag: Boolean = false) = Button(
```
Comment (LG95) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ComponentScreenBuilder.kt:110]:
```suggestion
    private fun createMenu(text: String, path: String) = Button(
```

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ComponentScreenBuilder.kt [hunk 5]
```diff
@@ -39,26 +39,44 @@ import br.com.zup.beagle.sample.constants.SCREEN_LIST_VIEW_ENDPOINT
...
+                                )
+                            )
+                        )
+                )
+        ) else NavigationBar(
+            title = "Choose a ${if (aqFlag) "Test" else "Sample"} Component",
             showBackButton = true
-        ),
+        )
+        ,
```
Comment (LG95) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ComponentScreenBuilder.kt:79]: Please simplify these redundant ifs.

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ImageRemoteScreenBuilder.kt [hunk 6]
```diff
@@ -40,10 +40,10 @@ import br.com.zup.beagle.widget.ui.ImagePath.Local
 import br.com.zup.beagle.widget.ui.ImagePath.Remote
 import br.com.zup.beagle.widget.ui.Text
 
-object NetworkImageScreenBuilder : ScreenBuilder {
+object ImageRemoteScreenBuilder : ScreenBuilder {
```
Comment (LG95) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/builder/ImageRemoteScreenBuilder.kt:43]: No `qaFlag`?

File: backend/sample/spring/src/main/kotlin/br/com/zup/beagle/sample/spring/service/SampleButtonService.kt [hunk 7]
```diff
@@ -21,5 +21,5 @@ import org.springframework.stereotype.Service
 
 @Service
 class SampleButtonService {
-    fun createButtonView() = ButtonScreenBuilder
-}
+    fun createButtonView(x:Boolean) = ButtonScreenBuilder(x)
```
Comment (LG95) [inline backend/sample/spring/src/main/kotlin/br/com/zup/beagle/sample/spring/service/SampleButtonService.kt:24]:
```suggestion
    fun createButtonView(qaFlag:Boolean) = ButtonScreenBuilder(qaFlag)
```

File: backend/sample/spring/src/main/kotlin/br/com/zup/beagle/sample/spring/service/SafeAreaService.kt [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.sample.spring.service
+
+import br.com.zup.beagle.sample.builder.SafeAreaBuilder
+import org.springframework.stereotype.Service
+
+@Service
+class SafeAreaService {
```
Comment (LG95) [inline backend/sample/spring/src/main/kotlin/br/com/zup/beagle/sample/spring/service/SafeAreaService.kt:23]: Please remove unused class.

File: backend/sample/micronaut/src/main/kotlin/br/com/zup/beagle/sample/micronaut/service/SafeAreaService.kt [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.sample.micronaut.service
+
+import br.com.zup.beagle.sample.builder.SafeAreaBuilder
+import javax.inject.Singleton
+
+@Singleton
+class SafeAreaService {
```
Comment (LG95) [inline backend/sample/micronaut/src/main/kotlin/br/com/zup/beagle/sample/micronaut/service/SafeAreaService.kt:23]: Where is this used?

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/constants/PathConstants.kt [hunk 10]
```diff
@@ -16,51 +16,54 @@
...
 const val CUSTOM_WIDGET_ENDPOINT = "/custom"
 
 const val PATH_SCREEN_DEEP_LINK_ENDPOINT = "screen-deep-link"
 
 const val PATH_LAZY_COMPONENT_ENDPOINT = "/text-lazy-component"
 
-const val CUSTOM_PLATFORM_SAMPLE_ENDPOINT = "/sample/platform/custom"
-const val PLATFORM_SAMPLE_ENDPOINT = "/sample/platform"
+const val CUSTOM_PLATFORM_SAMPLE_ENDPOINT = "/sample-platform-custom/$QAFLAG_PLACEHOLDER"
+const val PLATFORM_SAMPLE_ENDPOINT = "/sample-platform/$QAFLAG_PLACEHOLDER"
```
Comment (LG95) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/constants/PathConstants.kt:66]:
```suggestion
const val CUSTOM_PLATFORM_SAMPLE_ENDPOINT = "/sample/platform/custom/$QAFLAG_PLACEHOLDER"
const val PLATFORM_SAMPLE_ENDPOINT = "/sample/platform/$QAFLAG_PLACEHOLDER"
```

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/constants/PathConstants.kt [hunk 11]
```diff
@@ -16,51 +16,54 @@
...
+const val SCREEN_TEXT_ENDPOINT = "/text/$QAFLAG_PLACEHOLDER"
+const val SCREEN_IMAGE_ENDPOINT = "/image/$QAFLAG_PLACEHOLDER"
+const val SCREEN_ANALYTICS_ENDPOINT = "/analytics/$QAFLAG_PLACEHOLDER"
+const val SCREEN_TAB_VIEW_ENDPOINT = "/tab-view/$QAFLAG_PLACEHOLDER"
+const val SCREEN_LIST_VIEW_ENDPOINT = "/list-view/$QAFLAG_PLACEHOLDER"
+const val SCREEN_SCROLL_VIEW_ENDPOINT = "/scroll-view-screen/$QAFLAG_PLACEHOLDER"
+const val SCREEN_PAGE_VIEW_ENDPOINT = "/page-view-screen/$QAFLAG_PLACEHOLDER"
+const val SCREEN_ACTION_ENDPOINT = "/action/$QAFLAG_PLACEHOLDER"
+const val SCREEN_FORM_ENDPOINT = "/sample/form/$QAFLAG_PLACEHOLDER"
+const val SCREEN_LAZY_COMPONENT_ENDPOINT = "/lazy-component/c"
```
Comment (LG95) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/constants/PathConstants.kt:40]: Why the `/c`?

File: backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/constants/PathConstants.kt [hunk 12]
```diff
@@ -16,51 +16,54 @@
 
 package br.com.zup.beagle.sample.constants
 
-
+const val QAFLAG = "qaFlag"
+const val QAFLAG_PLACEHOLDER = "{$QAFLAG}"
```
Comment (LG95) [inline backend/sample/core/src/main/kotlin/br/com/zup/beagle/sample/constants/PathConstants.kt:20]: Please use `_` to separate words, such as `QA_FLAG`.

File: android/sample/src/main/java/br/com/zup/beagle/sample/MainActivity.kt [hunk 13]
```diff
@@ -49,6 +49,12 @@ class MainActivity : AppCompatActivity() {
             title = "Beagle Sample"
             elevation = 4.0f.dp()
         }
+        startActivity(BeagleActivity.newIntent(
```
Comment (uziasferreirazup) [inline android/sample/src/main/java/br/com/zup/beagle/sample/MainActivity.kt:52]: why you start activity when open the app?

**Comment by uziasferreirazup** [comment]
hey @luisgustavozup about this feature the team foundation create a meet to talk on monday, temporary I will close this PR to wait this meet

**Pull request closed without merging**